etna/pipeline/hierarchical_pipeline.py

```diff
@@ -0,0 +1,156 @@
+            Number of folds to use in the backtest for prediction interval estimation
+
+        Returns
+        -------
+        :
+            Dataset with predictions at the source level
+        """
+        forecast = super().forecast(prediction_interval=prediction_interval, quantiles=quantiles, n_folds=n_folds)
+        target_columns = tuple(get_target_with_quantiles(columns=forecast.columns))
+
+        hierarchical_forecast = TSDataset(
+            df=forecast[..., target_columns],
+            freq=forecast.freq,
+            df_exog=forecast.df_exog,
+            known_future=forecast.known_future,
+            hierarchical_structure=self.ts.hierarchical_structure,  # type: ignore
+        )
+        return hierarchical_forecast
+
+    def forecast(
+        self, prediction_interval: bool = False, quantiles: Sequence[float] = (0.025, 0.975), n_folds: int = 3
+    ) -> TSDataset:
+        """Make a prediction for target at the source level of hierarchy and make reconciliation to target level.
+
+        Parameters
+        ----------
+        prediction_interval:
+            If True returns prediction interval for forecast
+        quantiles:
+            Levels of prediction distribution. By default 2.5% and 97.5% taken to form a 95% prediction interval
+        n_folds:
+            Number of folds to use in the backtest for prediction interval estimation
+
+        Returns
+        -------
+        :
+            Dataset with predictions at the target level of hierarchy.
+        """
+        forecast = self.raw_forecast(prediction_interval=prediction_interval, quantiles=quantiles, n_folds=n_folds)
+        forecast_reconciled = self.reconciliator.reconcile(forecast)
+        return forecast_reconciled
+
+    def _compute_metrics(
+        self, metrics: List[Metric], y_true: TSDataset, y_pred: TSDataset
+    ) -> Dict[str, Dict[str, float]]:
+        """Compute metrics for given y_true, y_pred."""
+        if y_true.current_df_level != self.reconciliator.target_level:
+            y_true = y_true.get_level_dataset(self.reconciliator.target_level)
+
+        if y_pred.current_df_level == self.reconciliator.source_level:
+            y_pred = self.reconciliator.reconcile(y_pred)
+
+        metrics_values: Dict[str, Dict[str, float]] = {}
+        for metric in metrics:
+            metrics_values[metric.name] = metric(y_true=y_true, y_pred=y_pred)  # type: ignore
+        return metrics_values
+
+    def _forecast_prediction_interval(
+        self, predictions: TSDataset, quantiles: Sequence[float], n_folds: int
+    ) -> TSDataset:
+        """Add prediction intervals to the forecasts."""
+        self.forecast, self.raw_forecast = self.raw_forecast, self.forecast  # type: ignore
+
+        if self.ts is None or self._fit_ts is None:
+            raise ValueError("Pipeline is not fitted! Fit the Pipeline before calling forecast method.")
+
+        # TODO: rework intervals estimation for `BottomUpReconciliator`
```

Comment:
It is still relevant?

Reply:
Yes, this TODO is for the future work, we added support of prediction intervals but it is not perfect solution, this TODO remind us to improve it.